### PR TITLE
[ebpf] Disable socket filter for CentOS

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Disable socket filter loading process (feeding bytecode to the eBPF VM) in certain cases.

### Motivation

Certain kernels/distros (RedHat and CentOS) don't support eBPF socket filter programs.

### CODE REVIEW

File: pkg/ebpf/tracer.go [hunk 1]
```diff
@@ -711,6 +711,11 @@ func SectionsFromConfig(c *Config) map[string]bpflib.SectionParams {
 		tcpCloseEventMap.sectionName(): {
 			MapMaxEntries: 1024,
 		},
-		"socket/dns_filter": {},
 	}
+
+	if isRHELOrCentos, _ := isRHELOrCentOS(); isRHELOrCentos {
```
Comment (leeavital) [inline pkg/ebpf/tracer.go]: Should we also take into account whether or not DNS collecting is enabled?

Comment (p-lambert) [reply pkg/ebpf/tracer.go]: 👍 will update

File: pkg/ebpf/tracer.go [hunk 2]
```diff
@@ -711,6 +711,11 @@ func SectionsFromConfig(c *Config) map[string]bpflib.SectionParams {
 		tcpCloseEventMap.sectionName(): {
 			MapMaxEntries: 1024,
 		},
-		"socket/dns_filter": {},
```
Comment (sunhay) [inline pkg/ebpf/tracer.go]:
```
isRHELOrCentos, _ := isRHELOrCentOS();

return map[string]bpflib.SectionParams{
		connMap.sectionName(): {
			MapMaxEntries: int(c.MaxTrackedConnections),
		},
		tcpStatsMap.sectionName(): {
			MapMaxEntries: int(c.MaxTrackedConnections),
		},
		portBindingsMap.sectionName(): {
			MapMaxEntries: int(c.MaxTrackedConnections),
		},
		tcpCloseEventMap.sectionName(): {
			MapMaxEntries: 1024,
		},
		"socket/dns_filter": {
		    Disabled: isRHELOrCentos,
		},
	}
```

Comment (sunhay) [reply pkg/ebpf/tracer.go]: This seems more straightforward, no?

File: pkg/ebpf/tracer.go (hunk 1 above)
Comment (sunhay) [reply pkg/ebpf/tracer.go]: +1 here

File: pkg/ebpf/tracer.go (hunk 2 above)
Comment (p-lambert) [reply pkg/ebpf/tracer.go]: 👍

**Review by leeavital — DISMISSED** [review]

File: pkg/ebpf/tracer.go [hunk 3]
```diff
@@ -149,7 +149,7 @@ func NewTracer(config *Config) (*Tracer, error) {
 	}
 
 	var reverseDNS ReverseDNS = nullReverseDNS{}
-	if config.DNSInspection {
+	if shouldEnableSocketFilter(config) {
```
Comment (sunhay) [inline pkg/ebpf/tracer.go]: You can call `shouldEnableSocketFilter()` once above and use here here + pass it into `SectionsFromConfig()` as a param. 

Also `isRHELOrCentOS` is already called in this function below so perhaps we bring that up as well to reduce the number of calls to `getPlatform()` (which needs to go through a number of things to figure out the platform...)

**Review by sunhay — DISMISSED** [review]

**Review by sunhay — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**